apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -557,6 +566,25 @@ where
         tracing::info!("{}", stats);
         tracing::info!("{}", stats.format_tx_executed());
 
+        // Update the MASP commitment tree anchor if the tree was updated
```

Comment:
What is the purpose of writing to one key in the write log then writing the anchor to another key? Why not just write to the anchor key directly?

Reply:
I'm not sure I'm getting your point. The masp tx appends to the merkle tree in storage and here we just update the anchor at the end of the block. We could make the tx update the anchor itself and update the vp to validate it but to my understanding it would be of little benefit because no one could use those intermediate anchors. This way we can do the update only once and without the need for validation.

Reply:
They could technically use it in several transactions sent in sequence, but this is fine for now I think.